packages/gatsby-plugin-preload-fonts/README.md

```diff
@@ -0,0 +1,110 @@
+# gatsby-plugin-preload-fonts
+
+This plugin preloads all necessary fonts per route to decrease time to first
+meaningful paint. It works really well with services like Google Fonts that
+link to stylesheets that in turn link to font files.
+
+## Install
+
+```
+npm install --save gatsby-plugin-preload-fonts
+```
+
+Once you've installed the plugin, you'll want to add the included script
+to your project's scripts in `package.json`.
+
+```json
+{
+  "scripts": {
+    "preload-fonts": "gatsby-preload-fonts"
+  }
+}
+```
+
+## How to use
+
+In your `gatsby-config.js`:
+
+```javascript
+module.exports = {
+  plugins: [`gatsby-plugin-preload-fonts`],
+}
+```
+
+Before building your application, you will need to generate a font asset map
+using the included `gatsby-preload-fonts` script.
+
+```
+npm run preload-fonts
+```
+
+You'll need to run this script again after adding routes to your application
+(including new dynamic routes generated by source plugins), as well as any time
+you add new font assets or stylesheets that link to font assets.
+
+## Explanation of method
+
+A common pattern in modern web development is to link to a remote stylesheet that
+in turn links to other remote font assets. If it takes too long for the font to
+resolve/download, typography will either not show up or flash in a different
+system font before swapping out for the correct font.
+
+This can be avoided by preloading font assets on the initial page load instead of
+waiting for them to resolve indirectly. In order to do this, `<link rel="preload">`
+tags will need to be added for each font asset required on each route at build time.
+
+To accomplish this, this plugin runs in two steps:
+
+1. Scrape each application route to determine what font assets are ultimately required.
+2. Inject `<link rel="preload">` tags for each asset at build time.
+
+The included `gatsby-preload-fonts` script spins up a [puppeteer](https://github.com/GoogleChrome/puppeteer) instance and visits each of your application's
+routes to determine the required font assets. Once it has done so, it caches
```

Comment:
How is the caching done? My assumption is the script would write out a file which would get checked in via Git.

Reply:
This would let someone run this in e.g. a CI server (which we should include instructions for)

Reply:
The generated cache file `font-preload-cache.json` will be stored in the project root. In order to run this during CI, the user should be able to simply add `npm run preload-fonts`. Is that worth adding to the readme?

Reply:
yeah! That'd be great. A section recommending people set this up in CI would be helpful. Thanks!